ras/ras.py

```diff
@@ -60,28 +91,77 @@ def run_cmd_out(cmd):
                                      ignore_status=True,
                                      sudo=True).decode("utf-8").strip()
 
-    @skipIf(IS_POWER_NV or IS_KVM_GUEST, "This test is not supported on KVM guest or PowerNV platform")
+    def test(self):
+        """
+        For upstream target download and compile source code of
+        powerpc-utils, ppc64-diag and lsvpd
+        """
+
+        if self.run_type == 'upstream':
+            # Fetch and compile powerpc-utils source code
+            tarball = self.fetch_asset('ppcutil.zip',
+                                       locations=[self.ppcutils_url],
+                                       expire='7d')
+            archive.extract(tarball, self.workdir)
+            self.sourcedir = os.path.join(self.workdir, 'powerpc-utils-master')
+            os.chdir(self.sourcedir)
+            process.run("./autogen.sh", shell=True, verbose=True)
+            process.run("./configure", verbose=True)
+            build.make(self.sourcedir)
+            build.make(self.sourcedir, extra_args='install')
+            # Fetch and compile ppc64-diag source code
+            tarball = self.fetch_asset('ppcdiag.zip',
+                                       locations=[self.ppcdiag_url],
+                                       expire='7d')
+            archive.extract(tarball, self.workdir)
+            self.sourcedir = os.path.join(self.workdir, 'ppc64-diag-master')
+            os.chdir(self.sourcedir)
+            process.run("./autogen.sh", shell=True, verbose=True)
+            process.run("./configure", verbose=True)
+            build.make(self.sourcedir)
+            build.make(self.sourcedir, extra_args='install')
+            # Fetch and compile lsvpd source code
+            tarball = self.fetch_asset('lsvpd.zip',
+                                       locations=[self.lsvpd_url],
+                                       expire='7d')
+            archive.extract(tarball, self.workdir)
+            self.sourcedir = os.path.join(self.workdir, 'lsvpd-master')
+            os.chdir(self.sourcedir)
+            process.run("./bootstrap.sh", shell=True, verbose=True)
+            process.run("./configure", verbose=True, ignore_status=False)
+            build.make(self.sourcedir)
+            build.make(self.sourcedir, extra_args='install')
+            # For SUSE /sbin takes precedence over /usr/local/sbin.
```

Comment:
thinking as we clubbing all upstream repo tools as seeing upstream always in development so it can break while build process, can we handle this or have sperate test cases on these cases

Reply:
Yes, I can add checks for compile errors. 

Other issue I ran into is various tools are currently clubbed into a single py file. Adding upstream support will require downloading source code for all three packages in every single py file (ras.py, ras_extended.py and ras_tools.py)
This unnecessarily complicates the code.

I will work on re-writing and separating the tools so that all tools belonging to one package can be consolidated into a single test case.

Will close this PR for now and will send a separate one.